### ISSUE TITLE
NumPy array dtype may not be found when converting to Image

### ISSUE THREAD

**Issue by dzenanz** [InsightSoftwareConsortium/ITK#3544]
An issue described and fixed in SimpleITK https://github.com/SimpleITK/SimpleITK/issues/1685 might affect ITK Python too.

**Comment by tbirdso** [comment]
I am able to reproduce the issue in the most recent ITK pre-release:

```
Python 3.8.5 (default, Sep  3 2020, 21:29:08) [MSC v.1916 64 bit (AMD64)] :: Anaconda, Inc. on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import itk
>>> import numpy as np
>>> img = np.zeros([3,4,5],dtype=np.dtype('uintc'))
>>> img
array([[[0, 0, 0, 0, 0],
        [0, 0, 0, 0, 0],
        [0, 0, 0, 0, 0],
        [0, 0, 0, 0, 0]],

       [[0, 0, 0, 0, 0],
        [0, 0, 0, 0, 0],
        [0, 0, 0, 0, 0],
        [0, 0, 0, 0, 0]],

       [[0, 0, 0, 0, 0],
        [0, 0, 0, 0, 0],
        [0, 0, 0, 0, 0],
        [0, 0, 0, 0, 0]]], dtype=uint32)
>>> itk.GetImageFromArray(img)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\tom.birdsong\Anaconda3\envs\venv-itk\lib\site-packages\itk\support\extras.py", line 408, in GetImageFromArray
    return _GetImageFromArray(arr, "GetImageFromArray", is_vector, ttype)
  File "C:\Users\tom.birdsong\Anaconda3\envs\venv-itk\lib\site-packages\itk\support\extras.py", line 374, in _GetImageFromArray
    PixelType = _get_itk_pixelid(arr)
  File "C:\Users\tom.birdsong\Anaconda3\envs\venv-itk\lib\site-packages\itk\support\extras.py", line 292, in _get_itk_pixelid
    raise e
  File "C:\Users\tom.birdsong\Anaconda3\envs\venv-itk\lib\site-packages\itk\support\extras.py", line 287, in _get_itk_pixelid
    return _np_itk[numpy_array_type.dtype.type]
KeyError: <class 'numpy.uintc'
```

Will investigate applying the SimpleITK fix in ITK.

**Comment by yoda-vid** [comment]
I've encountered this issue recently with ITK 5.4.0, tested on Windows with Python 3.9 and 3.10. I saw that the PR fixing this issue (#3730) was merged into a `numpy-dtypes` branch, and I could not find the changes in [`master`](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Modules/Bridge/NumPy/wrapping/PyBuffer.i.init). Was this intentional, or is the fix meant to be in `master`?

Thanks for your help!
